### PR TITLE
add logger

### PR DESCRIPTION
**Pull request by stanislavstrelnikov**
little improvements

### CODE REVIEW

File: src/_util/logger.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { log } from "../deps.ts";
+
+export const logger = log.getLogger();
```
Comment (irustm) [inline src/_util/logger.ts:3]: move this file to a folder `src/utils/`

File: src/renderer/redirect.ts [hunk 2]
```diff
@@ -1,23 +1,16 @@
 import { ActionResult } from '../models/response.ts';
 
-export function Redirect(url: string): ActionResult {
+export function Redirect(url: string, permanent: boolean = false): ActionResult {
```
Comment (irustm) [inline src/renderer/redirect.ts:3]: Do not touch this file, it is necessary to implement each redirect function https://httpstatuses.com/.

File: src/renderer/redirect.ts [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
-    headers.append("Location", url);
-    
-    return {
-        status: 301,
-        headers,
-        __isActionResult: true,
-      };
-}    
\ No newline at end of file
+export function PermanentRedirect(url: string) {
```
Comment (irustm) [inline src/renderer/redirect.ts:14]: like this https://docs.microsoft.com/en-gb/dotnet/api/system.web.httpresponse.redirectpermanent?view=netframework-4.8

File: src/route/get-hooks.ts [hunk 4]
```diff
@@ -42,20 +38,19 @@ function getGrouped<TState,TPayload>(hooks: HookMetadataArgs<TState,TPayload>[],
   const controllerHooks: HookMetadataArgs<TState,TPayload>[] = [];
   const actionHooks: HookMetadataArgs<TState,TPayload>[] = [];
 
-  for (let hook of hooks) {
-    if(hook.object === action.areaObject) {
+  hooks.forEach((hook) => {
```
Comment (irustm) [inline src/route/get-hooks.ts:41]: is not necessary, because there may be async functions inside.

File: src/route/get-hooks.ts [hunk 5]
```diff
@@ -43,17 +39,16 @@ function getGrouped<TState,TPayload>(hooks: HookMetadataArgs<TState,TPayload>[],
   const actionHooks: HookMetadataArgs<TState,TPayload>[] = [];
 
   for (let hook of hooks) {
-    if(hook.object === action.areaObject) {
+    if (hook.object === action.areaObject) {
       areaHooks.push(hook);
-      continue;
```
Comment (irustm) [inline src/route/get-hooks.ts:48]: why was it removed?

`continue;`

Comment (stanislavstrelnikov) [reply src/route/get-hooks.ts:48]: > why was it removed?
> 
> `continue;`

my bad, fixed it

**Comment by irustm** [comment]
LGTM

Thanks!

**Comment by irustm** [comment]
I'll revert concole.log, 

because logger conflict with di container:

```
error: Uncaught Error: Another accept task is ongoing
    at unwrapResponse ($deno$/ops/dispatch_json.ts:43:11)
    at Object.sendAsync ($deno$/ops/dispatch_json.ts:98:10)
    at async ListenerImpl.accept ($deno$/net.ts:61:17)
    at async Server.acceptConnAndIterateHttpRequests (https://deno.land/std@0.56.0/http/server.ts:212:14)
    at async MuxAsyncIterator.callIteratorNext (https://deno.land/std@0.56.0/async/mux_async_iterator.ts:28:29)
```

**Pull request merged**